src/ngraph/frontend/onnx_import/op/lstm.cpp

```diff
@@ -0,0 +1,371 @@
+                                           "from 1, while direction attribute set to 'forward'.";
+                                    ng_in.second = reshape::squeeze(ng_in.second);
+                                }
+                            }
+                        }
+                    }
+
+                    ~LSTMNode() {}
+                    NodeVector run()
+                    {
+                        // ------ VARIABLE'S NAMES AND ACRONYM DEFINITIONS ------
+                        // The names used below are analogous to the one used in ONNX documentation.
+                        //
+                        // ------ INPUTS ------
+                        // X - The input tensor. [seq_length, batch_size, input_size]
+                        // W - The weight tensor. [num_directions, 4*hidden_size, input_size]
+                        // R - The recurrence weight tensor. [num_directions, 4*hidden_size, hidden_size]
+                        // B - The bias tensor for input gate. [num_directions, 8*hidden_size]
+                        // P - The weight tensor forr peepholes. [num_directions, 3*hidde_size]
+                        // ------ ACRONYMS ------
+                        // i - input gate
+                        // o - output gate
+                        // f - forget gate
+                        // c - cell gate
+                        // t - time step (t-1 means previous time step)
+                        // ------ VARIABLE NAMES ------
+                        // W       - W parameter weight matrix for input, output, forget, and
+                        //           cell gates.
+                        // R       - R recurrence weight matrix for input, output, forget, and
+                        //           cell gates.
+                        // Wb      - W bias vectors for input, output, forget, and cell gates.
+                        // Rb      - R bias vectors for input, output, forget, and cell gates.
+                        // b_W_R   - Bias vectors for input, output, forget, and cell gates.
+                        //           Concatenation of `[Wb, Rb]`.
+                        // p_[iof] - P peephole weight vector for respectively: input, output,
+                        //           and forget gates.
+                        // H_t     - Hidden state vector at current time step.
+                        // C_t     - Cell state vector at current time step.
+                        // h_list  - The list of hidden states at all processed time steps.
+                        //
+                        // Xt_W    - Input sequence multiplied by weights tensor at current time
+                        //           step.
+                        // Ht_W    - Hidden state multiplied by weights tensor at current time step.
+
+                        NodeVector p_iof = reshape::split(m_input_map["P"], 3);
+                        NgraphNodePtr p_i = p_iof.at(0);
```

Comment:
If I use plain `ngraph::Node` I get errors since all nGraph operators requires inputs as `std::shared_ptr<>`s. I've changed this into: `const auto& p_i = p_iof.at(0);`